### PR TITLE
Fix some audio mixer data races

### PR DESCRIPTION
**Pull request by sethalves**
- Fix some audio mixer data races

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- run the servers from this PR
- get a group of people together and test audio

File: assignment-client/src/AssignmentClient.cpp [hunk 1]
```diff
@@ -53,10 +53,14 @@ AssignmentClient::AssignmentClient(Assignment::Type requestAssignmentType, QStri
...
     DependencyManager::set<AccountManager>();
- 
+
     auto scriptableAvatar = DependencyManager::set<ScriptableAvatar>();
     auto addressManager = DependencyManager::set<AddressManager>();
-    auto scriptEngines = DependencyManager::set<ScriptEngines>();
+
+    if (requestAssignmentType != Assignment::AvatarMixerType &&
+        requestAssignmentType != Assignment::AudioMixerType) {
+        auto scriptEngines = DependencyManager::set<ScriptEngines>();
```
Comment (Atlante45) [inline assignment-client/src/AssignmentClient.cpp]: Isn't `AgentType` the only one to have a `ScriptEngine`?
Might as well do:
```cpp
if (requestAssignmentType != Assignment::AgentType) {
    auto scriptEngines = DependencyManager::set<ScriptEngines>();
}
```

File: assignment-client/src/AssignmentClient.cpp [hunk 2]
```diff
@@ -53,10 +53,14 @@ AssignmentClient::AssignmentClient(Assignment::Type requestAssignmentType, QStri
     QSettings::setDefaultFormat(QSettings::IniFormat);
 
     DependencyManager::set<AccountManager>();
- 
+
     auto scriptableAvatar = DependencyManager::set<ScriptableAvatar>();
     auto addressManager = DependencyManager::set<AddressManager>();
-    auto scriptEngines = DependencyManager::set<ScriptEngines>();
+
+    if (requestAssignmentType != Assignment::AvatarMixerType &&
```
Comment (ZappoMan) [inline assignment-client/src/AssignmentClient.cpp]: I'm pretty sure we only need this for agents... and not any of the other types

Comment (Atlante45) [inline assignment-client/src/AssignmentClient.cpp]: Arf, you beat me to it!

File: assignment-client/src/AssignmentClient.cpp (hunk 1 above)
Comment (sethalves) [inline assignment-client/src/AssignmentClient.cpp]: FINE!  I'll adjust

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0c6dedff4588b6e256bd26516977e82b71d34f42 available at: https://gist.github.com/3cf96301bc9a8b086db37001e0ef0d5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8843-0c6dedf.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8843-0c6dedf.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9cb4756fce54c42634c401e0179b8f9ba1ad034c available at: https://gist.github.com/3cf96301bc9a8b086db37001e0ef0d5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8843-9cb4756.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8843-9cb4756.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 150c9b4bf8961ec4b8268e2d53f0841be7ae3923 available at: https://gist.github.com/3cf96301bc9a8b086db37001e0ef0d5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8843-150c9b4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8843-150c9b4.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 72998b190c701ee73a0391716fb6cf1a9e34769f available at: https://gist.github.com/3cf96301bc9a8b086db37001e0ef0d5f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8843-72998b1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8843-72998b1.dmg

**Comment by ZappoMan** [comment]
What's our test plan on this? Just a smoke test?

**Comment by thoys** [comment]
@ZappoMan [see above](https://github.com/highfidelity/hifi/pull/8843#issuecomment-254632413)

**Comment by thoys** [comment]
## QA Testing notes

Couldn't repro the issue with 9 Interface'd avatars. The fix doesn't seem to impact the way audio works, or any performance, so we mutually decided to QA PASS this, and hopefully it will come to a crash-less Friday `Weekly Meet-up` this week.

QA PASS

**Pull request merged**